include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,473 @@
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_ANGLE_HPP
+#define SFML_ANGLE_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Represents an angle value.
+///
+////////////////////////////////////////////////////////////
+class Angle
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Sets the angle value to zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Angle();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in degrees
+    ///
+    /// \return Angle in degrees
+    ///
+    /// \see asRadians
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asDegrees() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in radians
+    ///
+    /// \return Angle in radians
+    ///
+    /// \see asDegrees
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asRadians() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [-180°, 180°)
+    ///
+    /// \return Angle wrapped to [-180°, 180°)
+    ///
+    /// \see wrappedTo360
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo180() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [0°, 360°)
+    ///
+    /// \return Angle wrapped to [0°, 360°)
```

Comment:
I think, we should explain what the `[x, y)` range means

Reply:
Should we add a line below this that reads something like this?
```
Inclusive of 0, exclusive of 360
```

Reply:
I think only one of the two should be adjusted, the `\brief` could maybe stay.

Maybe we can directly use the relational operators?
```cpp
    ////////////////////////////////////////////////////////////
    /// \brief Return the angle wrapped to the range [0°, 360°)
    ///
    /// \return Angle >= 0° and < 360°
```

Reply:
I'd have gone the other way.
The return description can be rather crisp, while the function description should try and clarify in more details what stuff means and how they can/should be used etc.

Reply:
I think it makes sense to have `0° <= angle < 360°` somewhere.

Reply:
> The return description can be rather crisp, while the function description should try and clarify in more details what stuff means and how they can/should be used etc.

Doesn't `\brief` literally mean "short/concise"? I think it's important that this one is relatively fast to read, as it will be listed among other function declarations.

But I'm totally fine with something like
```cpp
    ////////////////////////////////////////////////////////////
    /// \brief Wrap to a range, so that 0° <= angle < 360°
    ///
    /// \return Angle in [0°, 360°), which represents a rotation equivalent to *this.
```

Reply:
One could also add it in between the two, which then is a normal function description. The `\brief` is indeed listed in the function list overview.

![image](https://user-images.githubusercontent.com/920861/151943610-2d06bc5a-82c4-4da8-9bbb-abcc1693f50e.png)

I think both brief and return text can use the same notation and then if someone doesn't understand the notation, they should be able to find more details in the description.

Reply:
Yes, but we should also avoid duplicating information, it just wastes the reader's time. (I know we're doing it everywhere else, and still think it's bad practice).

For me,
```cpp
\return Angle in [0°, 360°), which represents a rotation equivalent to *this.
```
seems to be clear enough -- but if we want to elaborate, we could also do it in `\return` directly. The middle section is not strictly necessary.

Reply:
@Bromeon @eXpl0it3r Now that the two of you have approved this PR, can I resolve this last remaining issue?